iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -376,19 +377,19 @@ public override async Task SendTwinPatchAsync(TwinCollection reportedProperties,
             try
             {
                 await EnableTwinPatchAsync(cancellationToken).ConfigureAwait(false);
-                await RoundTripTwinMessage(AmqpTwinMessageType.Patch, reportedProperties, cancellationToken).ConfigureAwait(false);
+                await RoundTripTwinMessageAsync(AmqpTwinMessageType.Patch, reportedProperties, cancellationToken).ConfigureAwait(false);
             }
             finally
             {
                 Logging.Exit(this, reportedProperties, cancellationToken, $"{nameof(SendTwinPatchAsync)}");
             }
         }
 
-        private async Task<Twin> RoundTripTwinMessage(AmqpTwinMessageType amqpTwinMessageType, TwinCollection reportedProperties, CancellationToken cancellationToken)
+        private async Task<Twin> RoundTripTwinMessageAsync(AmqpTwinMessageType amqpTwinMessageType, TwinCollection reportedProperties, CancellationToken cancellationToken)
         {
-            Logging.Enter(this, cancellationToken, $"{nameof(RoundTripTwinMessage)}");
+            Logging.Enter(this, cancellationToken, $"{nameof(RoundTripTwinMessageAsync)}");
 
-            string correlationId = amqpTwinMessageType + Guid.NewGuid().ToString();
+            string correlationId = amqpTwinMessageType + Guid.NewGuid().ToString().Substring(amqpTwinMessageType.ToString().Length);
```

Comment:
@drwill-ms If there is a better way to overlay Amqp message type such as "Get" on guid let me know.

Reply:
I've thought about it and can't come up with anything better than what you have. Perhaps you should make a private method out of it, though. Also, I wonder about caching the amqpTwinMessageType length so you don't have to call ToString and get the length each time.

Reply:
I'm tempted to just go back to simple concat to keep it simple, and not worry about 3-5 extra characters.

Reply:
I think the same for simplicity, will change it back